File: app/src/main/java/com/eshopthis/finds/Activities/ArtsActivity.java
package com.eshopthis.finds.Activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import com.eshopthis.finds.Adapter.Adapter;
import com.eshopthis.finds.R;
import com.eshopthis.finds.db.Model;

import java.util.ArrayList;
import java.util.List;

public class ArtsActivity extends AppCompatActivity {

    private RecyclerView recyclerViewArts;
    private Adapter adapter;
    private List<Model> dataList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_arts);

        recyclerViewArts = findViewById(R.id.recyclerViewArts);
        recyclerViewArts.setLayoutManager(new GridLayoutManager(this, 2)); // Adjust the number of columns as needed

        dataList = new ArrayList<>();
        adapter = new Adapter(this, dataList);
        recyclerViewArts.setAdapter(adapter);

        loadBooks();
    }

    private void loadBooks() {
        dataList.clear();
        dataList.add(new Model("1", "Art Book 1", "$19.99", "Description for Art Book 1", R.drawable.arts));
        dataList.add(new Model("2", "Art Book 2", "$29.99", "Description for Art Book 2", R.drawable.arts));
        // Ensure you have drawable resources named art_book_1 and art_book_2 in your drawable folder
        // Add more books as needed
        adapter.notifyDataSetChanged();
    }
}

File: app/src/main/java/com/eshopthis/finds/Activities/ArtsActivity.kt
package com.eshopthis.finds.Activities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.eshopthis.finds.Adapter.Adapter
import com.eshopthis.finds.R
import com.eshopthis.finds.data.Model

class ArtsActivity : AppCompatActivity() {

    private lateinit var recyclerViewArts: RecyclerView
    private lateinit var adapter: Adapter
    private val dataList = mutableListOf<Model>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_arts)

        recyclerViewArts = findViewById(R.id.recyclerViewArts)
        recyclerViewArts.layoutManager = GridLayoutManager(this, 2)

        adapter = Adapter(this, dataList)
        recyclerViewArts.adapter = adapter

        loadBooks()
    }

    private fun loadBooks() {
        dataList.clear()
        dataList.add(Model("1", "Art Book 1", "$19.99", "Description for Art Book 1", R.drawable.arts))
        dataList.add(Model("2", "Art Book 2", "$29.99", "Description for Art Book 2", R.drawable.arts))
        adapter.notifyDataSetChanged()
    }
}